Index: git/src/gui/imagewidget.cpp
===================================================================
--- git.orig/src/gui/imagewidget.cpp	2023-09-24 09:15:27.409136180 +0200
+++ git/src/gui/imagewidget.cpp	2023-09-24 10:37:18.873897782 +0200
@@ -501,11 +501,7 @@
 void ImageWidget::cancelScan() {
 #ifdef HAVE_KSANE
   if(m_saneWidget) {
-#if KSANE_VERSION < QT_VERSION_CHECK(22,4,0)
     m_saneWidget->scanCancel();
-#else
-    m_saneWidget->cancelScan();
-#endif
   }
 #endif
 }
