Index: git/src/gui/imagewidget.cpp
===================================================================
--- git.orig/src/gui/imagewidget.cpp	2025-11-27 05:42:43.251849464 +0100
+++ git/src/gui/imagewidget.cpp	2025-11-27 05:42:43.244849465 +0100
@@ -502,11 +502,7 @@
 void ImageWidget::cancelScan() {
 #ifdef HAVE_KSANE
   if(m_saneWidget) {
-#if KSANE_VERSION < QT_VERSION_CHECK(22,4,0)
     m_saneWidget->scanCancel();
-#else
-    m_saneWidget->cancelScan();
-#endif
   }
 #endif
 }
