Index: git/src/translators/xslthandler.cpp
===================================================================
--- git.orig/src/translators/xslthandler.cpp	2013-03-11 19:17:22.000000000 +0100
+++ git/src/translators/xslthandler.cpp	2013-03-11 19:18:41.000000000 +0100
@@ -80,9 +80,7 @@
 int XSLTHandler::s_initCount = 0;
 
 XSLTHandler::XSLTHandler(const QByteArray& xsltFile_) :
-    m_stylesheet(0),
-    m_docIn(0),
-    m_docOut(0) {
+    m_stylesheet(0) {
   init();
   QByteArray file = QUrl::toPercentEncoding(QString::fromLocal8Bit(xsltFile_));
   if(!file.isEmpty()) {
@@ -97,9 +95,7 @@
 }
 
 XSLTHandler::XSLTHandler(const KUrl& xsltURL_) :
-    m_stylesheet(0),
-    m_docIn(0),
-    m_docOut(0) {
+    m_stylesheet(0) {
   init();
   if(xsltURL_.isValid() && xsltURL_.isLocalFile()) {
     xmlDocPtr xsltDoc = xmlReadFile(xsltURL_.encodedPathAndQuery().toUtf8(), NULL, xslt_options);
@@ -113,9 +109,7 @@
 }
 
 XSLTHandler::XSLTHandler(const QDomDocument& xsltDoc_, const QByteArray& xsltFile_, bool translate_) :
-    m_stylesheet(0),
-    m_docIn(0),
-    m_docOut(0) {
+    m_stylesheet(0) {
   init();
   QByteArray file = QUrl::toPercentEncoding(QString::fromLocal8Bit(xsltFile_));
   if(!xsltDoc_.isNull() && !file.isEmpty()) {
@@ -128,14 +122,6 @@
     xsltFreeStylesheet(m_stylesheet);
   }
 
-  if(m_docIn) {
-    xmlFreeDoc(m_docIn);
-  }
-
-  if(m_docOut) {
-    xmlFreeDoc(m_docOut);
-  }
-
   --s_initCount;
   if(s_initCount == 0) {
     xsltUnregisterExtModule(EXSLT_STRINGS_NAMESPACE);
@@ -230,12 +216,13 @@
     return QString();
   }
 
+  xmlDocPtr m_docIn;
   m_docIn = xmlReadDoc(reinterpret_cast<xmlChar*>(text_.toUtf8().data()), NULL, NULL, xml_options);
 
-  return process();
+  return process(m_docIn);
 }
 
-QString XSLTHandler::process() {
+QString XSLTHandler::process(xmlDocPtr m_docIn) {
   if(!m_docIn) {
     myDebug() << "error parsing input string!";
     return QString();
@@ -252,10 +239,15 @@
     i += 2;
   }
   // returns NULL on error
+  xmlDocPtr m_docOut;
   m_docOut = xsltApplyStylesheet(m_stylesheet, m_docIn, params.data());
   for(int i = 0; i < 2*m_params.count(); ++i) {
     delete[] params[i];
   }
+  
+  xmlFreeDoc(m_docIn);
+  m_docIn = 0;
+
   if(!m_docOut) {
     myDebug() << "error applying stylesheet!";
     return QString();
@@ -268,6 +260,10 @@
       myDebug() << "error saving output buffer!";
     }
   }
+
+  xmlFreeDoc(m_docOut);
+  m_docOut = 0;
+  
   return output.result();
 }
 
Index: git/src/translators/xslthandler.h
===================================================================
--- git.orig/src/translators/xslthandler.h	2013-03-11 19:17:22.000000000 +0100
+++ git/src/translators/xslthandler.h	2013-03-11 19:18:41.000000000 +0100
@@ -108,11 +108,9 @@
 
 private:
   void init();
-  QString process();
+  QString process(xmlDocPtr m_docIn);
 
   xsltStylesheetPtr m_stylesheet;
-  xmlDocPtr m_docIn;
-  xmlDocPtr m_docOut;
 
   QHash<QByteArray, QByteArray> m_params;
 
